### ISSUE TITLE
fork bomb when running xalt_extract_record with very liberal XALT configuration

### ISSUE THREAD

**Issue by boegel** [xalt/xalt#29]
We're using `xalt_extract_record` as a way to verify that the XALT configuration we've obtained is working.

While testing the XALT installation process, I was providing a very "liberal" XALT configuration file, which boils down to creating records for pretty much anything.

XALT was configured with:

```
./configure --prefix=/software/XALT/2.8.1  --build=x86_64-pc-linux-gnu  --host=x86_64-pc-linux-gnu  --with-siteControlledPrefix=yes  --with-config=$HOME/xalt.cfg  --with-syshostConfig=hardcode:test --with-transmission=syslog  --with-MySQL=no  LDFLAGS="${LDFLAGS} -static-libstdc++ -static-libgcc"
```

and the contents of `$HOME/xalt.cfg`:

```python
# capture executables on all nodes
hostname_patterns = [
  ['KEEP', '.*']
]
# don't track XALT tools
path_patterns = [
  ['SKIP', r'.*xalt.*'],
  ['SKIP', r'.*\/XALT\/.*']
]
```

Once XALT was installed, running `xalt_extract_record` seemed to hang, and after a couple dozen seconds there were over 120k XALT messages in syslog:

```
$ grep XALT_LOGGING /var/log/messages | wc -l
121479
```

Here's part of the process tree:

```
kehoste 15034  0.0  0.0 125684  1552 pts/0    S+   21:38   0:00              \_ /bin/bash /software/XALT/2.8.1/bin/xalt_configuration_report
kehoste 15035  0.0  0.0 125684   536 pts/0    S+   21:38   0:00                  \_ /bin/bash /software/XALT/2.8.1/bin/xalt_configuration_report
kehoste 15036  0.0  0.0 125688   924 pts/0    S+   21:38   0:00                      \_ /bin/bash /software/XALT/2.8.1/bin/xalt_configuration_report
kehoste 15045  0.0  0.0  43012  2448 pts/0    S+   21:38   0:00                          \_ /software/XALT/2.8.1/libexec/xalt_run_submission --interfaceV 5 --pid 15034 --ppid 14721 --start 1590176332.5907 --end 1590176332.6892 --exec /usr/bin/bash --ntasks 1 --kind scalar --build_UUID --return_UUID --prob 1 --ngpus 0 --watermark RkFMU0U= -- WyIvYmluL2Jhc2giLCIvdXNlci9nZW50LzQwMC92c2M0MDAyMy9lYl9zY3JhdGNoL0NPNy9oYXN3ZWxsLWliL3NvZnR3YXJlL1hBTFQvMi44LjEvYmluL3hhbHRfY29uZmlndXJhdGlvbl9yZXBvcnQiXQ==
kehoste 15055  0.0  0.0 125688  1732 pts/0    S+   21:38   0:00                              \_ sh -c PATH=/usr/bin:/bin objdump -s -j .xalt "/usr/bin/bash" 2> /dev/null
kehoste 15065  0.0  0.0  43012  2448 pts/0    S+   21:38   0:00                                  \_ /software/XALT/2.8.1/libexec/xalt_run_submission --interfaceV 5 --pid 15055 --ppid 15045 --start 1590176332.7243 --end 1590176332.7538 --exec /usr/bin/bash --ntasks 1 --kind scalar --build_UUID --return_UUID --prob 1 --ngpus 0 --watermark RkFMU0U= -- WyJzaCIsIi1jIiwiUEFUSD0vdXNyL2JpbjovYmluIG9iamR1bXAgLXMgLWogLnhhbHQgXCIvdXNyL2Jpbi9iYXNoXCIgMj4gL2Rldi9udWxsIl0=
kehoste 15075  0.0  0.0 125688  1736 pts/0    S+   21:38   0:00                                      \_ sh -c PATH=/usr/bin:/bin objdump -s -j .xalt "/usr/bin/bash" 2> /dev/null
kehoste 15089  0.0  0.0  43012  2452 pts/0    S+   21:38   0:00                                          \_ /software/XALT/2.8.1/libexec/xalt_run_submission --interfaceV 5 --pid 15075 --ppid 15065 --start 1590176332.7874 --end 1590176332.8371 --exec /usr/bin/bash --ntasks 1 --kind scalar --build_UUID --return_UUID --prob 1 --ngpus 0 --watermark RkFMU0U= -- WyJzaCIsIi1jIiwiUEFUSD0vdXNyL2JpbjovYmluIG9iamR1bXAgLXMgLWogLnhhbHQgXCIvdXNyL2Jpbi9iYXNoXCIgMj4gL2Rldi9udWxsIl0=
kehoste 15099  0.0  0.0 125688  1736 pts/0    S+   21:38   0:00                                              \_ sh -c PATH=/usr/bin:/bin objdump -s -j .xalt "/usr/bin/bash" 2> /dev/null
kehoste 15110  0.0  0.0  43012  2456 pts/0    S+   21:38   0:00                                                  \_ /software/XALT/2.8.1/libexec/xalt_run_submission --interfaceV 5 --pid 15099 --ppid 15089 --start 1590176332.8743 --end 1590176332.9041 --exec /usr/bin/bash --ntasks 1 --kind scalar --build_UUID --return_UUID --prob 1 --ngpus 0 --watermark RkFMU0U= -- WyJzaCIsIi1jIiwiUEFUSD0vdXNyL2JpbjovYmluIG9iamR1bXAgLXMgLWogLnhhbHQgXCIvdXNyL2Jpbi9iYXNoXCIgMj4gL2Rldi9udWxsIl0=
kehoste 15120  0.0  0.0 125688  1736 pts/0    S+   21:38   0:00                                                      \_ sh -c PATH=/usr/bin:/bin objdump -s -j .xalt "/usr/bin/bash" 2> /dev/null
```

It's not clear to me why the path filtering isn't working to avoid this, or why this even occurs in the first place; XALT should definitely avoid creating a fork bomb if possible...

**Comment by rtmclay** [comment]
Thanks for the example showing the problem.  I have found the problem.  I now set XALT_EXECUTABLE_TRACKING=no when doing system calls and popen calls.  This now prevents fork bombs in my tests. Please test XALT 2.8.4 to see if works for you.

**Comment by boegel** [comment]
The problem is indeed fixed with XALT 2.8.4, thanks for the fix!
